### ISSUE TITLE
insert as select does not work with parameters mapping

### ISSUE THREAD

**Issue by shm-dmitry** [crate/crate#10653]
**CrateDB version**: 4.2.6

**Problem description**:
Mapping parameters for prepared SQL statement failed for insert-as-select query

**Steps to reproduce**:

`create table test (x int)`
>> `Table TEST created.`

`select x from test where x = :param` (and set :param to any value)
>> ok

`insert into test (x) select x from test where x = 123 `
>> ok

`insert into test (x) select x from test where x = :param ` (and set :param to any value)
>> **`SQL Error: Индекс колонки вне диапазона: 1. Допустимые значения: 1..0`** (`Column index out of range: 1. Valid values: 1..0` - sorry, I dont know how to change language to engish)

**Comment by mfussenegger** [comment]
CrateDB doesn't support named parameters like `:param` on the server side. Either `?` or numbered placeholders like `$1` can be used.

Otherwise the client needs to take care of replacing them. Which client are you using and how are you setting up the parameter values?

**Comment by shm-dmitry** [comment]
@mfussenegger , In code I used `?`. 
I dont use named parameters in code.
I used it in sample because it simpler to check in SQL Developer.

**Comment by mfussenegger** [comment]
@shm-dmitry I'm afraid we'll need some more information to look into this?

Could you elaborate and what exactly you're doing with which tools and post the outcome you're seeing and the outcome you'd expect?

**Comment by shm-dmitry** [comment]
@mfussenegger 
I using wrapper for JDBC (similar to JDBCTemplate)
All parameters there passed using `?`

My SQL looks like a (sorry - it was minor issue for me, there was more important problems, and I forget to save original SQL. At now I cant found it in logs)
```
INSERT INTO table1 (col1, col2, col3, col4)
SELECT a, b, ?, c
FROM table2
WHERE d = ?
```

And this request executes via 
```
		try (var conn = DriverManager.getConnection("jdbc:postgresql://localhost:5432/", "crate", "")) {
			try (var stm = conn.prepareStatement("insert into test (x) select x from test where x = ?")) {
				stm.setObject(1, 1);
				stm.execute();
			}
		}
```

I tryed to reproduce this now - and all works fine ...
Maybe it related to #10650  ? I dont know.
In about a week I will try to reproduce #10650 on test cluster, and I will check this also.

Sorry for disturbing

**Comment by shm-dmitry** [comment]
@mfussenegger , I reproduced this on 4.2.7

Sorry for issue description - it is incorrect.

Steps for reproduce - execute this java code (checked with both crate and PG JDBC drivers).
On 4.1.8 all works fine.

```
	public void test2() throws Exception {
		try (var conn = DriverManager.getConnection("jdbc:crate://ev-crate00:5432/", "crate", "")) {
			try (var stmt = conn.prepareStatement(
					"create table if not exists test.test (catalog_id string, is_active boolean, source string, user_id string, user_order int)")) {
				stmt.execute();
			}

			try (var stmt = conn.prepareStatement("create table if not exists test.test1 (id string, arr array(int))")) {
				stmt.execute();
			}

			try (var stmt = conn.prepareStatement(
					"INSERT INTO test.test (catalog_id , is_active , source , user_id , user_order ) SELECT ID, true, 'autotest', ?, 0 from test.test1 where 110 = ANY(arr) and 152 = any(arr) LIMIT ?")) {
				stmt.setObject(1, 7);
				stmt.setObject(2, 12);

				stmt.execute();
			}

		}
	}
``` 

You will receive error like
```
io.crate.shade.org.postgresql.util.PSQLException: ERROR: SQLParseException: Couldn't create executionContexts from
NodeOperations: [2=>NodeOp{ phase{id=2/fetchPhase, nodes=[55NoEaEMSrm5J12vr0DSuA, HnoECOnhRs2C_U6z0Hwa5g, XnIi2RdYS2-mPNw60dWLNA]}, downstreamNodes=[], downstreamPhase=2147483647, downstreamInputId=0}, 0=>NodeOp{ phase{id=0/collect, nodes=[55NoEaEMSrm5J12vr0DSuA, HnoECOnhRs2C_U6z0Hwa5g, XnIi2RdYS2-mPNw60dWLNA], dist=BROADCAST}, downstreamNodes=[_response], downstreamPhase=1, downstreamInputId=0}]
Leafs: [MergePhase{executionPhaseId=1, name=mergeOnHandler, projections=[TopNProjection{outputs=[INPUT(0), INPUT(1), INPUT(2), INPUT(3), INPUT(4)], limit=12, offset=0}, io.crate.execution.dsl.projection.FetchProjection@2, io.crate.execution.dsl.projection.EvalProjection@a5d7f7a3, io.crate.execution.dsl.projection.ColumnIndexWriterProjection@2dcf6bf1], outputTypes=[bigint], jobId=c022047f-9c6c-4c10-b10f-38f707b76568, numUpstreams=3, numInputs=1, nodeOperations=[55NoEaEMSrm5J12vr0DSuA], inputTypes=[bigint, boolean, text, integer, integer], orderBy=null}]
target-sources: [1=>[0]]
original-error: UnsupportedOperationException: Can't handle Symbol [ParameterSymbol: $1]
	at io.crate.shade.org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at io.crate.shade.org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at io.crate.shade.org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at io.crate.shade.org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:442)
	at io.crate.shade.org.postgresql.jdbc.PgStatement.execute(PgStatement.java:366)
	at io.crate.shade.org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:145)
	at io.crate.shade.org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:134)
```

**Comment by mfussenegger** [comment]
@shm-dmitry thank you for the updated test case. We came up with a fix (https://github.com/crate/crate/pull/10742) which will be released with 4.3.2
